packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -230,8 +248,45 @@ export const useOrganizationSwitcherContext = () => {
   const navigateCreateOrganization = () => navigate(ctx.createOrganizationUrl || displayConfig.createOrganizationUrl);
   const navigateOrganizationProfile = () =>
     navigate(ctx.organizationProfileUrl || displayConfig.organizationProfileUrl);
-  const navigateAfterSwitchOrganization = () =>
-    ctx.afterSwitchOrganizationUrl ? navigate(ctx.afterSwitchOrganizationUrl) : Promise.resolve();
+
+  const navigateAfterSwitchOrganization = ({
```

Comment:
Are we going to keep the internal name or should we rename this to `navigateAfterSelectOrganization`?

Reply:
Not sure if that would be helpful. 

If we split this to 2 function then we are exposing the conditional logic inside of `handlePersonalWorkspaceClicked` in order to have the same result. If in any case we want to create more than 2  handlePersonalWorkspaceClicked functions in more than 2 components this will result to more duplicate code.

Reply:
Sorry If I wasn't clear - I'm just saying that we probably want to rename this to better match the new public API